Export MostReadItem from psammead-most-read (#2491)
BBC-archive/psammead: packages/components/psammead-most-read/README.md

## Description

> The `MostReadItem` component is designed to display the most read articles given a designated period of time which is dependent on service. The component comprises of a `MostReadTitle`, a `MostReadList` which is a grid containing `MostReadItems`. A `MostReadItem` comprises of a numerical counter representing it's ranking and a link to the article.

Reviewer: ```suggestion
The `MostReadItem` component is designed to display the most read articles given a designated period of time which is dependent on service. The component comprises of a `MostReadTitle`, a `MostReadList` which is a grid containing `MostReadItems`. A `MostReadItem` comprises of a numerical counter representing its ranking and a link to the article.
```

## Installation

`npm install @bbc/psammead-most-read/esm/item`

## Props

> 

Reviewer 2: ```suggestion
### MostReadItem props
```

<!-- prettier-ignore -->
| Argument | Type | Required | Default | Example |
| -------- | ---- | -------- | ------- | ------- |
| item | object | yes | N/A | `{ title: 'This is a item', href: 'https://www.bbc.com' }` |

> | count | string | yes | N/A | `'rtl'`  |

Reviewer 3: Is this prop name correct? I don't think the naming of this prop matches the values I'd expect it to take?

> | count | string | yes | N/A | `'rtl'`  |

Reviewer 2: ```suggestion
| count | string | yes | N/A | `'2'`  |
```

| script | object | yes | N/A | `{ canon: { groupA: { fontSize: '28', lineHeight: '32',}, groupB: { fontSize: '32', lineHeight: '36', }, groupD: { fontSize: '44', lineHeight: '48', }, }, trafalgar: { groupA: { fontSize: '20', lineHeight: '24', }, groupB: { fontSize: '24', lineHeight: '28', }, groupD: { fontSize: '32', lineHeight: '36', }, }, }` |
| service | string | yes | N/A | `'news'` |
| dir | string | no | `'ltr'` | `'rtl'`  |

> | lastUpdated | node | no | N/A | `<time>12 March 2019</time>` |

Reviewer: ```suggestion
| lastUpdated | node | no | null | `<time>12 March 2019</time>` |
```

## Usage

> A typical use-case of this component is as displayed below. It contains a count and an info elements. The info element is a link which point to the corresponding article while the count is a number that represent the position of the info element on the list of info elements.

Reviewer: ```suggestion
A typical use-case of this component is as displayed below. It contains a count and an info element. The info element is a link which points to the corresponding article while the count is a number that represents the position of the info element on the list of info elements.
```

## Description

> The `MostRead` component is designed to be used on the `article` pages, which are pages that display articles/stories. It's designed to have a numerical count and a link to another article.

Reviewer 4: Added a little more detail to the description. Might be a little verbose but we can cut the last bit detailing how the component is made based on individual parts.

```suggestion
The `MostRead` component is designed to display the most read articles given a designated period of time which is dependent on service. The component comprises of a `MostReadTitle`, a `MostReadList` which is a grid containing `MostReadItems`. A `MostReadItem` comprises of a numerical counter representing it's ranking and a link to the article.
```

| count | string | yes | N/A | `'rtl'`  |
| script | object | yes | N/A | `{ canon: { groupA: { fontSize: '28', lineHeight: '32',}, groupB: { fontSize: '32', lineHeight: '36', }, groupD: { fontSize: '44', lineHeight: '48', }, }, trafalgar: { groupA: { fontSize: '20', lineHeight: '24', }, groupB: { fontSize: '24', lineHeight: '28', }, groupD: { fontSize: '32', lineHeight: '36', }, }, }` |
| service | string | yes | N/A | `'news'` |
| dir | string | no | `'ltr'` | `'rtl'`  |

## Usage

A typical use-case of this component is as displayed below. It contains a count and an info elements. The info element is a link which point to the corresponding article while the count is a number that represent the position of the info element on the list of info elements.

```jsx
import React from 'react';
import MostRead from '@bbc/psammead-most-read';

const item = {
    header: 'Cranberries singer O'Riordan died by drowning,
    href: 'https://www.bbc.com'
}

<MostRead item={item} count="1" service="news" dir="ltr" />;

```

![most-read-single-item](documentation/most-read-single-item.png)

### When to use this component

> <!-- ### When not to use this component -->

Reviewer 4: ```suggestion
This component is to be used on `article` pages.
```

| service | string | yes | N/A | `'news'` |

## StyledCountSpan Usage

```jsx
import React from 'react';
import { latin } from '@bbc/gel-foundations/scripts';
import { StyledCountSpan } from '@bbc/psammead-most-read/esm/item';

<StyledCountSpan script={latin} service="news">10</StyledCountSpan>;

```

### When to use this component

This component is to be used on `article` pages.

### Accessibility notes

Currently this component is in alpha. This is because it has not yet been tested with various assistive technologies. After it has had an accessibility swarm, this will be published under a standard version.

## Roadmap

Currently, this package only exports the `MostReadItem` component. We will add a wrapper that will export a list of most read components using `@bbc/psammead-grid` to display the most read items on a grid.
=======
The `MostRead` component is designed to display the most read articles given a designated period of time which is dependent on service. The component comprises of a `MostReadTitle`, a `MostReadList` which is a grid containing a `MostReadCount/Rank` (name subject to change) and a `MostReadItem`.

> ## Installation

Reviewer 2: You can delete this section (Installation / MostReadTitle Props table), since it is duplicated with the installation steps above.
